cmd/zpool/zpool_vdev.c

```diff
@@ -1610,6 +1610,7 @@ construct_spec(nvlist_t *props, int argc, char **argv)
 	if (seen_logs && nlogs == 0) {
 		(void) fprintf(stderr, gettext("invalid vdev specification: "
 		    "log requires at least 1 device\n"));
+		free(top);
```

Comment:
I only looked at this quickly, but shouldn't "spares" and "l2cache" be free()d here too?